test/client/application-cache.spec.js

```diff
@@ -0,0 +1,220 @@
+import applicationCache from '../../public/js/src/module/application-cache';
+import events from '../../public/js/src/module/event';
+import settings from '../../public/js/src/module/settings';
+
+describe('Application cache initialization (offline service worker registration)', () => {
```

Comment:
Nice to see some tests that help describe the intended behavior!

Form resource updates/load aren't tested anywhere else, right? Maybe they're not candidates for this kind of test because the form itself is still using IndexedDB-based caching?

Reply:
From @eyelidlessness: we're doing this as a first step because it's a huge improvement but there's an additional change that will come soon to save the form resources in the browser cache rather than indexeddb so let's hold off on adding the testing until then.